pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -183,6 +183,22 @@ func (e *eksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 					return cluster, err
 				}
 			}
+			addNgMessage := "Cannot deploy agent without nodegroups. Add a nodegroup."
+			if len(cluster.Spec.EKSConfig.NodeGroups) == 0 {
+				cluster, err = e.setFalse(cluster, apimgmtv3.ClusterConditionWaiting, addNgMessage)
+				if err != nil {
+					return cluster, err
+				}
+			} else {
+				if apimgmtv3.ClusterConditionWaiting.GetMessage(cluster) == addNgMessage {
```

Comment:
This seems somewhat brittle. What if it gets into a state where the first part of conditional executes and error message is set and then rancher gets restarted. Would first part execute again and update error message ? Otherwise this is good case for a constant error code.

Reply:
The error would be updated. The SetFalse function checks if the current message matches the inputted message and updates it if not.